File: abc/abc136/abc136_c.cs
// https://atcoder.jp/contests/abc136/submissions/8119634

using System;
using System.Linq;

namespace CSharp
{
    class Program
    {
        public static void Main()
        {
            var N = int.Parse(Console.ReadLine());
            var H = Console.ReadLine().Split(' ').Select(int.Parse).ToArray();

            var result = true;
            for (int i = N - 1; i > 0; i--)
            {
                if (H[i] + 1 < H[i - 1])
                {
                    result = false;
                    break;
                }
                else if (H[i] + 1 == H[i - 1])
                {
                    H[i - 1]--;
                }
            }
            Console.WriteLine(result ? "Yes" : "No");
        }
    }
}

File: abc/abc136/abc136_c.fs
// https://atcoder.jp/contests/abc136/submissions/11990618

let N = stdin.ReadLine() |> int
let H = stdin.ReadLine().Split() |> Array.map int

let rec solver i =
    if i = 0 then "Yes"
    elif H.[i] + 1 < H.[i - 1] then "No"
    elif H.[i] + 1 = H.[i - 1] then
        H.[i - 1] <- H.[i - 1] - 1
        solver (i - 1)
    else solver (i - 1)

solver (N - 1) |> stdout.WriteLine
